packages/terra-clinical-result/src/ClinicalResultBloodPressure.jsx

```diff
@@ -0,0 +1,292 @@
+    ...customProps
+  } = props;
+
+  let clinicalResultBloodPressureDisplay = null;
+
+  if (hasResultError) {
+    clinicalResultBloodPressureDisplay = <ResultError />;
+  } else if (hasResultNoData) {
+    clinicalResultBloodPressureDisplay = <NoData />;
+  } else {
+    const CompareTemplate = (s = null, d = null) => ({
+      systolic: s,
+      diastolic: d,
+    });
+    const compareUnits = new CompareTemplate();
+    const compareConceptDisplays = new CompareTemplate();
+    const compareDatetimeDisplays = new CompareTemplate();
+    const compareStatusIsInError = new CompareTemplate();
+
+    let hasModifiedIcon = false;
+    let hasCommentIcon = false;
+    let hasUnverifiedIcon = false;
+
+    let iconGroupDisplayElement = null;
+    let conceptDisplayElement = null;
+    let datetimeDisplayElement = null;
+    const decoratedResultDisplay = [];
+
+    const hasSystolic = resultData.systolic;
+    const noDataSystolic = (hasSystolic && resultData.systolic.resultNoData === true);
+    if (hasSystolic && !noDataSystolic) {
+      const {
+        result = null,
+        status = null,
+        conceptDisplay = null,
+        datetimeDisplay = null,
+        isModified = null,
+        hasComment = null,
+        isUnverified = null,
+      } = hasSystolic;
+
+      if (!isEmpty(result.unit)) { compareUnits.systolic = result.unit.trim().toLowerCase(); }
+      if (!isEmpty(status)) { compareStatusIsInError.systolic = checkIsStatusInError(hasSystolic); }
+      if (!isEmpty(conceptDisplay)) { compareConceptDisplays.systolic = conceptDisplay.trim().toLowerCase(); }
+      if (!isEmpty(datetimeDisplay)) { compareDatetimeDisplays.systolic = datetimeDisplay.trim().toLowerCase(); }
+      if (!isEmpty(isModified)) { hasModifiedIcon = isModified; }
+      if (!isEmpty(hasComment)) { hasCommentIcon = hasComment; }
+      if (!isEmpty(isUnverified)) { hasUnverifiedIcon = isUnverified; }
+    }
+
+    const hasDiastolic = resultData.diastolic;
+    const noDataDiastolic = (hasDiastolic && resultData.diastolic.resultNoData === true);
+    if (hasDiastolic && !noDataDiastolic) {
+      const {
+        result = null,
+        status = null,
+        conceptDisplay = null,
+        datetimeDisplay = null,
+        isModified = null,
+        hasComment = null,
+        isUnverified = null,
+      } = hasDiastolic;
+
+      if (!isEmpty(result.unit)) { compareUnits.diastolic = result.unit.trim().toLowerCase(); }
+      if (!isEmpty(status)) { compareStatusIsInError.diastolic = checkIsStatusInError(hasDiastolic); }
+      if (!isEmpty(conceptDisplay)) { compareConceptDisplays.diastolic = conceptDisplay.trim().toLowerCase(); }
+      if (!isEmpty(datetimeDisplay)) { compareDatetimeDisplays.diastolic = datetimeDisplay.trim().toLowerCase(); }
+      if (!isEmpty(isModified)) { hasModifiedIcon = isModified; }
+      if (!isEmpty(hasComment)) { hasCommentIcon = hasComment; }
+      if (!isEmpty(isUnverified)) { hasUnverifiedIcon = isUnverified; }
```

Comment:
Is  all these checks needed for `hasModifiedIcon`, `hasCommentIcon` and `hasUnverifiedIcon`? It seems a little redundant to add an additional boolean check when we could use the prop value for the checks below.... In lines 205-208:
```diff
-      const modifiedIconElement = hasModifiedIcon && !hasUnverifiedIcon ? (<IconModified className={cx('icon-modified')} />) : null;
+      const modifiedIconElement = isModified && !isUnverified ? (<IconModified className={cx('icon-modified')} />) : null;
-      const commentIconElement = hasCommentIcon && !hasUnverifiedIcon ? (<IconComment className={cx('icon-comment')} />) : null;
+      const commentIconElement = hasComment && !isUnverified ? (<IconComment className={cx('icon-comment')} />) : null;
-      const unverifiedIconElement = hasUnverifiedIcon ? (<IconUnverified className={cx('icon-unverified')} />) : null;
+      const unverifiedIconElement = isUnverified ? (<IconUnverified className={cx('icon-unverified')} />) : null;
-      if (hasModifiedIcon || hasCommentIcon || hasUnverifiedIcon) {
+      if (isModified || hasComment || isUnverified) {
        iconGroupDisplayElement = (
          <React.Fragment>
            {modifiedIconElement}
            {commentIconElement}
            {unverifiedIconElement}
          </React.Fragment>
        );
      }

```